stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/images/CoilStreamImageLoader.kt

```diff
@@ -82,6 +95,59 @@ internal object CoilStreamImageLoader : StreamImageLoader {
         }
     }
 
+    override suspend fun loadAndResize(
+        target: ImageView,
+        data: Any?,
+        container: ViewGroup,
+        placeholderDrawable: Drawable?,
+        maxHeight: Int,
+        transformation: StreamImageLoader.ImageTransformation,
+        onStart: () -> Unit,
+        onComplete: () -> Unit,
+    ) {
+        val context = target.context
+
+        val drawable = withContext(DispatcherProvider.IO) {
+            val result = context.streamImageLoader.execute(
+                ImageRequest.Builder(context)
+                    .headers(imageHeadersProvider.getImageRequestHeaders().toHeaders())
+                    .data(data)
+                    .placeholder(placeholderDrawable)
+                    .applyTransformation(transformation)
+                    .listener(
+                        onStart = { onStart() },
+                        onCancel = { onComplete() },
+                        onError = { _, _ -> onComplete() },
+                        onSuccess = { _, _ -> onComplete() },
+                    )
+                    .build()
+            )
+
+            result.drawable
+        } ?: return
+
+        val widthToHeightRatio = (drawable.intrinsicWidth / drawable.intrinsicHeight.toFloat()).coerceAtMost(1.3f)
```

Comment:
Why do we use exactly 1.3 here?

Reply:
Honestly it's a bit of a magic number. It's hard for us to know and measure the maximum width of the parent so we cannot use that here, and we're using an arbitrary coefficient for the resize.

It's supposed to help us not scale things too much width wise, so that the container doesn't fall outside of its bounds.

I didn't have a better (or good) solution to this.

Reply:
I tried it without any coersion and none of the giphyes actually went outside of the parent.
I'm not sure if I just lucked out or if the container does well to respect parent's width but it's worth a shot.

Reply:
Can we move this to a constant?